src/Microsoft.AspNet.OData.Shared/Formatter/Deserialization/ODataResourceSetDeserializer.cs

```diff
@@ -147,10 +192,40 @@ public virtual IEnumerable ReadResourceSet(ODataResourceSetWrapper resourceSet,
                     Error.Format(SRResources.TypeCannotBeDeserialized, elementType.FullName()));
             }
 
-            foreach (ODataResourceWrapper resourceWrapper in resourceSet.Resources)
+            //Ideally we don't need to special case ResourceSetType.ResourceSet, since the code that handles a deltaresourcesetwrapper will also handle a ResourceSetWrapper, 
+            //but it may be more efficient for the common case.
+
+            if (resourceSet.ResourceSetType == ResourceSetType.ResourceSet)
             {
-                yield return deserializer.ReadInline(resourceWrapper, elementType, readContext);
+                foreach (ODataResourceWrapper resourceWrapper in resourceSet.Resources)
+                {
+                    yield return deserializer.ReadInline(resourceWrapper, elementType, readContext);
+                }
+            }
+            else
+            {
+                Type clrType = EdmLibHelpers.GetClrType(elementType, readContext.Model);                           
+
+                foreach (ODataItemBase odataItemBase in resourceSet.Resources)
+                {
+                    ODataResourceWrapper resourceWrapper = odataItemBase as ODataResourceWrapper;
+
+                    if (readContext.IsUntyped)
+                    {
+                        readContext.ResourceType = resourceWrapper.ResourceBase is ODataDeletedResource ? typeof(EdmDeltaDeletedEntityObject) : typeof(EdmDeltaEntityObject);
```

Comment:
should just use EdmEntityObject, not use EdmDeltaEntityObject?

Reply:
EdmDeltaEntityObject inherits from IEdmChangedObject  (edmchangedobjcoll is coll if idmchangedobject) , dut EdmEntity doesnt

---
In reply to: [604621960](https://github.com/OData/WebApi/pull/2455#discussion_r604621960) [](ancestors = 604621960)

Reply:
@sreejith -- can you resolve this issue if it's been addressed (I think it has, by implementing IEdmChangedObject on EdmEntityObject and using EdmEntityObject here?)

---
In reply to: [604695489](https://github.com/OData/WebApi/pull/2455#discussion_r604695489) [](ancestors = 604695489,604621960)